README.md

```diff
@@ -381,464 +607,224 @@ class MyBaseClass extends WidgetBase<WidgetProperties> {
+Extending the base class found in `meta/Base` will automatically add these hooks to the class instance as well as providing a `has` method:
 
-##### Applying a Theme
+* `has(key: string)` - A method that returns `true` if the DOM element with the passed key exists in the rendered DOM.
 
-Themeable widgets include an optional `theme` property which can be set to pass in a theme. Theme classes will override `baseClasses`. When a `theme` property is set or changed, the widgets `theme` classes will be regenerated and the widget invalidated such that it is redrawn. Themes are used to apply consistent styling across the widget codebase.
+Meta classes that require extra options should accept them in their methods.
 
-Usage Extending on the previous `tabPanel` example.
+```typescript
+import MetaBase from "@dojo/widget-core/meta/Base";
 
-``` css
-/* customTheme/tabPanel.css */
-.tabs {
-	background: green;
+interface IsTallMetaOptions {
+    minHeight: number;
 }
-```
-
-Import the theme and pass it to the widget via its `properties`. The theme classes will be automatically mixed into the widget and available via `this.classes`.
 
-```ts
-import * as customTheme from './themes/customTheme.css';
-
-interface MyThemeableWidgetProperties extends WidgetProperties, ThemeableProperties;
-
-class MyThemeableWidget extends ThemeableMixin(WidgetBase)<MyThemeableWidgetProperties> {
-	render: function () {
-		// Resulting widget will have green tabs instead of baseTheme red.
-		return w(TabPanel, { theme: customTheme });
-	}
+class IsTallMeta extends MetaBase {
+    isTall(key: string, { minHeight }: IsTallMetaOptions = { minHeight: 300 }): boolean {
+        this.requireNode(key);
+        const node = this.nodes.get(key);
+        if (node) {
+            return node.offsetHeight >= minHeight;
+        }
+        return false;
+    }
 }
 ```
 
-The theme can be applied to individual widgets or to a project and property passed down to its children.
-
-##### Injecting a theme
+### JSX Support
 
-The theming system supports injecting a theme that is configured externally to the usual mechanism of passing properties down the widget tree.
+In addition to creating widgets functionally using the `v()` and `w()` functions from `@dojo/widget-core/d`, Dojo 2 optionally supports the use of the `jsx` syntax known as [`tsx`](https://www.typescriptlang.org/docs/handbook/jsx.html) in TypeScript.
 
-This is done using a `ThemeInjectorContext` instance that is passed to the `Injector` mixin along with the `ThemeInjector` class. Once the theme injector is defined in the registry, the `theme` can be changed by calling the `ThemeInjectorContext#set(theme: any)` API on the instance of the injector context.
-
-```ts
-// Create the singleton injector context
-const themeInjectorContext = new ThemeInjectorContext(myTheme);
+To start to use `jsx` in your project the widgets need to be named with a `.tsx` extension and some configuration is required in the project's `tsconfig.json`:
```

Comment:
"to use `jsx` in your project the widgets" -> "using `jsx` in your project, widgets"